hochschule-fur-soziale-arbeit-fhnw.csl

```diff
@@ -0,0 +1,247 @@
+        <text variable="event-place" prefix=" " suffix="."/>
+      </if>
+    </choose>
+  </macro>
+  <!--Makro Darstellung der Herausgeber in Literaturverzeichnis-->
+  <macro name="container-contributors">
+    <choose>
+      <if type="chapter" match="any">
+        <group delimiter=". ">
+          <choose>
+            <if variable="author">
+              <names variable="editor" suffix=".">
+                <name delimiter="/" delimiter-precedes-last="never" initialize="false" initialize-with="." name-as-sort-order="all"/>
+                <label form="short" plural="always" prefix=" (" suffix=")"/>
+              </names>
+            </if>
+          </choose>
+          <choose>
+            <if variable="author editor" match="any">
+              <names variable="translator">
+                <name form="long" name-as-sort-order="all" and="text" delimiter="; " delimiter-precedes-last="never"/>
+                <label form="short" prefix=" (" suffix=")"/>
+              </names>
+            </if>
+          </choose>
+        </group>
+      </if>
+    </choose>
+  </macro>
+  <!--Makro Darstellung der Auflage im Literaturverzeichnis-->
+  <macro name="edition">
+    <choose>
+      <if variable="edition" match="any">
+        <text variable="edition" form="short" prefix=". " suffix=". Aufl. "/>
+      </if>
+    </choose>
+  </macro>
+  <!--Makro Darstellung des Verlags und Verlagsorts in Literaturverzeichnis-->
+  <macro name="publisher">
+    <choose>
+      <if type="book chapter report paper-conference" match="any">
+        <group delimiter=": ">
+          <text variable="publisher-place" prefix=" "/>
+          <text variable="publisher" suffix="."/>
+        </group>
+      </if>
+      <else-if type="thesis">
+        <group delimiter=". ">
+          <text variable="publisher" prefix=" "/>
+          <text variable="publisher-place" suffix="."/>
+        </group>
+      </else-if>
+    </choose>
+  </macro>
+  <!--Makro Darstellung Jahrgang, Ausgabe und Seitenzahl in Literaturverzeichnis-->
+  <macro name="locators">
+    <choose>
+      <if type="article article-journal article-magazine chapter">
+        <text variable="volume" prefix=" " suffix=". Jg."/>
+        <text variable="issue" prefix=" (" suffix=")."/>
+        <text variable="page" prefix=" S. " suffix=""/>
+      </if>
+    </choose>
+  </macro>
+  <!--Makro Darstellung URL und Zugriffsdatum in Literaturverzeichnis-->
+  <macro name="access">
+    <choose>
+      <if type="book chapter" match="none">
+        <choose>
+          <if variable="URL" match="any">
+            <text variable="URL" prefix="URL: "/>
+            <date variable="accessed" prefix=" [Zugriffsdatum: " suffix="].">
+              <date-part form="numeric-leading-zeros" name="day" suffix=". "/>
+              <date-part form="long" name="month" suffix=" "/>
+              <date-part name="year"/>
+            </date>
+          </if>
+          <else-if variable="DOI" match="any">
```

Comment:
Is this really what the guide wants? Preferring a URL over a DOI seems misguided to me. DOIs are permanent. URLs aren't necessarily.